ldms/src/ldmsd/ldmsd.c

```diff
@@ -2116,7 +2116,7 @@ int ldmsd_process_cmd_line_arg(char opt, char *value)
 		}
 		break;
 	case 't':
-		log_truncate = 1;
+		/* already done log_truncate = 1; */
```

Comment:
This will break the case that -t is given in configuration files, i.e., `ldmsd_process_cmd_line_arg()` is called by `__cmdline_options()` in ldmsd_request.c.

I believe an alternative solution is to call `ldmsd_open_log()` after processing the command line options in configuration files at line 2315 before the comment `/* Initialize LDMS */`. This could also avoid pre-scaning for `-t`.

Reply:
@nichamon Good catch. The tricky bit is that deferring open-log until after processing all the options (and emitting all the associated warnings/errors to something other than the -l specified log) means the user doesn't get the diagnostics about other arguments.
What do we do instead? A prescan for -t in both main and __cmdline_options()?